packages/api-v2/test/plan/plan.e2e-spec.ts

```diff
@@ -0,0 +1,183 @@
+import { INestApplication } from "@nestjs/common";
+import { Test, TestingModule } from "@nestjs/testing";
+import { Plan } from "../../src/plan/entities/plan.entity";
+import { Student } from "../../src/student/entities/student.entity";
+import * as request from "supertest";
+import { Connection } from "typeorm";
+import { AppModule } from "../../src/app.module";
+
+const testUser = {
+  fullName: "Tester",
+  nuid: "000000000",
+  email: "test-plan@gmail.com",
+  password: "1234567890",
+  academicYear: 2019,
+  graduateYear: 2023,
+  catalogYear: 2019,
+  major: "Computer Science",
+};
+
+const testUser2 = {
+  fullName: "Tester",
+  nuid: "000000000",
+  email: "test-plan2@gmail.com",
+  password: "1234567890",
+  academicYear: 2019,
+  graduateYear: 2023,
+  catalogYear: 2019,
+  major: "Computer Science",
+};
+
+const testPlan = {
+  name: "Test Plan",
+  schedule: {
+    years: [2019, 2020, 2021, 2022],
+    yearMap: {},
+  },
+  major: "Computer Science",
+  coopCycle: "4 year 2 co-ops",
+  concentration: "Artificial Intelligence",
+  catalogYear: 2019,
+  courseWarnings: [],
+  warnings: [],
+};
+
+describe("PlanController (e2e)", () => {
+  let app: INestApplication;
+  let jwtToken: string;
+  let connection: Connection;
+  let uuid: string;
+  let planID: number;
+
+  beforeEach(async () => {
+    const moduleFixture: TestingModule = await Test.createTestingModule({
+      imports: [AppModule],
+    }).compile();
+    app = moduleFixture.createNestApplication();
+
+    await app.init();
+    connection = app.get(Connection);
+
+    // create student
+    const res = await request(app.getHttpServer())
+      .post("/auth/register")
+      .send(testUser);
+
+    // save accessToken and user ID
+    jwtToken = res.body.accessToken;
```

Comment:
If I'm not wrong with Cindy's PR, we need to change this cuz the jwt won't be in the body anymore but will be attached as a cookie. I think when you merge/rebase with main, you prob have to update this portion